src/batcher/batcher.rs

```diff
@@ -0,0 +1,85 @@
+use crate::batcher::solve_pair::solve_pair;
+use crate::batcher::submit_solution::submit_solution;
+use crate::models::orderbook::OrderBook;
+use anyhow::Result;
+use ethcontract::web3::types::Address;
+use std::collections::HashMap;
+
+pub async fn batch_process(orderbook: OrderBook) -> Result<()> {
+    let tokens: Vec<Address> = orderbook
+        .orders
+        .read()
+        .keys()
+        .into_iter()
+        .map(|t| Address::from(t.as_fixed_bytes()))
+        .collect();
+
+    let token_pairs = get_token_pairs(&tokens);
+    for token_pair in token_pairs {
+        let best_match = solve_pair(
+            orderbook
```

Comment:
ditto. Would be nice if we can just get all sell orders for a given token pair from the model directly.